x509-cert/src/serial_number.rs

```diff
@@ -34,13 +34,15 @@ impl SerialNumber {
 
     /// Create a new [`SerialNumber`] from a byte slice.
     pub fn new(bytes: &[u8]) -> Result<Self> {
-        let inner = Int::new(bytes)?;
+        let inner = Uint::new(bytes)?;
 
         if inner.len() > SerialNumber::MAX_LEN {
             return Err(ErrorKind::Overlength.into());
         }
 
-        Ok(Self { inner })
+        Ok(Self {
+            inner: inner.into(),
+        })
```

Comment:
With this change, users will no longer be able to directly build a new negative serial number: the only way they'll be able to end up with a `SerialNumber` with a negative internal representation is by loading an existing one.